services/user/Chainmail/service/src/lib.rs

```diff
@@ -74,25 +68,61 @@ fn serve_rest_api(request: &HttpRequest) -> Option<HttpReply> {
+        let archived_requested = match params.get(&String::from("archived")) {
+            Some(arch) => arch == "true",
+            None => false,
+        };
+        println!("archived_requested: {}", archived_requested);
+
+        let mut where_clause_sender_receiver: String = String::from("");
         if s_opt.is_some() {
-            where_clause += s_clause.as_str();
+            where_clause_sender_receiver += s_clause.as_str();
         }
         if s_opt.is_some() && r_opt.is_some() {
-            where_clause += " AND ";
+            where_clause_sender_receiver += " AND ";
         }
         if r_opt.is_some() {
-            where_clause += r_clause.as_str();
+            where_clause_sender_receiver += r_clause.as_str();
         }
-
-        let mq = make_query(
-            request,
-            format!(
-                "SELECT * FROM \"history.chainmail.sent\" {} ORDER BY ROWID",
-                where_clause
-            ),
+        println!("where-clause: {}", where_clause_sender_receiver);
+
+        // let archived_msgs_query =     "SELECT DISTINCT sent.rowid as event_id,                   sent.* FROM \"history.chainmail.sent\" AS sent INNER JOIN \"history.chainmail.archive\" AS archive ON CONCAT(sent.receiver, sent.rowid) = archive.event_id";
+        // let not_archvied_msgs_query = "SELECT DISTINCT sent.rowid as event_id, archive.event_id, sent.* FROM \"history.chainmail.sent\" AS sent LEFT JOIN \"history.chainmail.archive\" AS archive ON CONCAT(sent.receiver, sent.rowid) = archive.event_id WHERE event_id IS NULL";
+        // Select from all sent emails *not archived* where receiver/send are as query params specify
+        let select_clause =
+            format!("DISTINCT archive.msg_id as archived_msg_id, sent.*, sent.rowid as msg_id");
+        let from_clause = format!("\"history.chainmail.sent\" AS sent LEFT JOIN \"history.chainmail.archive\" AS archive ON CONCAT(sent.receiver, sent.rowid) = archived_msg_id" );
+        let where_clause_archived_or_not = format!(
+            "archived_msg_id IS {} NULL",
+            if archived_requested { "NOT" } else { "" }
+        );
+        let order_by_clause = "sent.ROWID";
+
+        let sql_query_str = format!(
+            "SELECT {} FROM {} WHERE {} {} {} ORDER BY {}",
+            select_clause,
+            from_clause,
+            where_clause_archived_or_not,
+            if s_opt.is_some() || r_opt.is_some() {
+                "AND"
+            } else {
+                ""
+            },
+            where_clause_sender_receiver,
+            order_by_clause
         );
-        return REventsSvc::call().serveSys(mq);
+
+        println!("query: {}", sql_query_str);
+
+        let query_response = REventsSvc::call().sqlQuery(sql_query_str);
+        println!("query_response: {}", query_response);
+
+        return Some(HttpReply {
+            status: 200,
+            contentType: request.contentType.clone(),
```

Comment:
No `contentType` header is set on the HTTP request that originates from the `getJson` plugin request. You should probably just use "application/json" explicitly.